README.md

```diff
@@ -6,35 +6,133 @@ Exercism exercises in Haskell
-- `src/examples/success-foo/src/ModuleName.hs` is the source code of the sample solution.
-- `test/Tests.hs` is the test suite.
-- `HINTS.md` is an optional file containing instructions and/or hints.
+- `HINTS.md` is an optional file containing instructions and/or hints. It is used together with the respective `description.md` for the exercise from [x-common](https://github.com/exercism/x-common) to build the `README.md` file.
+- `src/ModuleName.hs` is a [stub solution](#stub-solution).
+- `examples/success-<name>/package.yaml` contains library dependencies for the [example solution](#example-solution). `<name>` is a unique name for the example - usually "standard" (as in `success-standard`), but it can be some other name in case of multiple example solutions.
+- `examples/success-<name>/src/ModuleName.hs` is the source code of the sample solution.
+- `test/Tests.hs` is the [test suite](#test-suite).
 
-### Running Tests
+### Writing an issue
+To report a bug you should [create an issue](https://help.github.com/articles/creating-an-issue/) [here](https://github.com/exercism/xhaskell/issue).
 
-In order to be accepted by Travis-CI, every exercise must compile without
-warnings and the example solution must pass the tests without failures.
+### Writing a pull request
+To fix a bug you should [create a pull request from a fork](https://help.github.com/articles/creating-a-pull-request-from-a-fork/) [here](https://github.com/exercism/xhaskell/pull). See also [here](https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#git-basics) for more information.
+
+### Development Dependencies
+You should have [Stack](http://docs.haskellstack.org/) installed in your system to make contributing to this repository easier.
+
+### Stub solution
+The stub solution must compile together with the test suite. It should be as general as possible in order to not exclude any possible solutions. It should take Haskell specifics into account (for example use `Maybe` instead of a dummy return value). It should not contain any comments (people might forget to remove them), you can use the hints file instead.
+
+### Example solution
+The example solution could be inspiration for other language implementors. It doesn't need to be perfect or very elegant. But it should be efficient enough for the test suite to finish in only a few seconds.
+
+### Test suite
+The test suite should be derived from the respective `x-common/exercises/<exercise-name>/canonical-data.json` and comply to some formatting and coding standards (to get an idea you may look at some of the existing tests).
+
+### Running Tests
+In order to be accepted by Travis-CI, every exercise must be registered in
+`config.json`, it must compile without warnings and the example solution must
+pass the tests without failures. Additionally the tests should not run longer than
+a few seconds.
 
 To test an exercise, first you need to rename the file `src/Example.hs`
```

Comment:
This section is completely outdated, because we changed a lot of thing in the past months. (#398)

- We don't have `Example.hs` anymore.
- We don't have an easy way to test an exercise.

We can fix that in another PR and maybe remove it from here now. At least until we come up with better instructions.